Plugins/Public/alley/PlayerRestrictions.cpp

```diff
@@ -1082,42 +1086,62 @@ void __stdcall SystemSwitchOutComplete(unsigned int iShip, unsigned int iClientI
 void JettisonCargo(unsigned int iClientID, struct XJettisonCargo const &jc)
 {
 	returncode = DEFAULT_RETURNCODE;
-	//int iSlotPlayer;
-
-	boolean matchFound = false;
-	for (list<EquipDesc>::iterator item = Players[iClientID].equipDescList.equip.begin(); item != Players[iClientID].equipDescList.equip.end(); item++)
+	
+	for (const EquipDesc& item : Players[iClientID].equipDescList.equip)
 	{
-		if (item->sID == jc.iSlot)
+		if (item.sID != jc.iSlot)
 		{
-			matchFound = true;
-			//PrintUserCmdText(iClientID, L"Slot match");
-			for (map<uint, string>::iterator i = notradelist.begin(); i != notradelist.end(); ++i)
-			{
-				if (i->first == item->iArchID)
-				{
-					returncode = SKIPPLUGINS_NOFUNCTIONCALL;
-					PrintUserCmdText(iClientID, L"ERR you can't jettison %s.", stows(i->second).c_str());
-				}
+			continue;
+		}
+
+		const auto& noTradeGood = notradelist.find(item.iArchID);
+		if (noTradeGood == notradelist.end()) {
+			return;
+		}
+
+		returncode = SKIPPLUGINS_NOFUNCTIONCALL;
+		const GoodInfo* gi = GoodList::find_by_id(noTradeGood->first);
+		wstring goodName = HkGetWStringFromIDS(gi->iIDSName).c_str();
+		if (noTradeGood->second == 0.0f)
+		{
+			PrintUserCmdText(iClientID, L"ERR you can't jettison %ls.", goodName.c_str());
+		}
+		else
+		{
+			uint amountToJettison = static_cast<uint>(jc.iCount * noTradeGood->second);
```

Comment:
so the config is supposed to be a float between 0 and 1 - if they configure it higher, or negative, we could have some issues here, right?

Reply:
Correct. I know that for zero specifically it still jettisons a single unit of cargo. Above 1 would allow to generate cargo on the spot. Will clamp those values on config load to 0 and 1 respectively.